tests/e2e/multicluster/verify-argocd/argocd_test.go

```
@@ -114,7 +123,103 @@ var _ = t.Describe("Multi Cluster Argo CD Validation", Label("f:platform-lcm.ins
 			}, waitTimeout, pollingInterval).Should(BeTrue())
 		})
 	})
+	t.Context("Token Update Tests", func() {
+		t.BeforeEach(func() {
+			os.Setenv(k8sutil.EnvVarTestKubeConfig, adminKubeconfig)
+		})
+		// Checks that the secret corresponding to the managed-secret in the cluster has both createdAt and ExpiredAt annotations
+		t.It("The expected secret currently contains both the createdAt and ExpiredAt annotations", func() {
+			secretName := fmt.Sprintf("%s-argocd-cluster-secret", managedClusterName)
+			Eventually(func() (bool, error) {
+				return verifyCreatedAtAndExpiresAtTimestampsExist(argoCDNamespace, secretName)
+			}, waitTimeout, pollingInterval).Should(BeTrue(), "Expected to find both Created and Expired At Annotations "+secretName)
+		})
+		// Tests that a new ArgoCD token is able to be created
+		t.It("A new ArgoCD token is able to be created through the Rancher API", func() {
+			Eventually(func() error {
+				httpClientForRancher, clusterID, APIAccessToken, err := getRequiredInfoToPreformTokenOperationsInRancherForArgoCD()
+				if err != nil {
+					pkg.Log(pkg.Error, "Error getting the required information to preform Token Operations in Rancher")
+					return err
+				}
+				createdTimeStampForNewTokenCreated, err = pkg.AddAccessTokenToRancherForLoggedInUser(httpClientForRancher, adminKubeconfig, clusterID, ttl, APIAccessToken, *t.Logs)
+				if err != nil {
+					pkg.Log(pkg.Error, "Error creating New Token")
+					return err
+				}
+				return err
+
+			}, waitTimeout, pollingInterval).ShouldNot(HaveOccurred(), "Expected to Be Able To Create a Token through the API")
```

I think you can change this to a `ShouldNot(BeEmpty())` and return the `createdTimeStampForNewTokenCreated`

---

Typically for these tests, we only want to return the error if it is the only information we have. Otherwise, we can look at a variable and return the error as the second parameter. This gives us more information on test failures.

---

Thanks!